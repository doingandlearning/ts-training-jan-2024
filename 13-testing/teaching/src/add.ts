export function add(a: number, b: number): number {
  if (typeof a !== "number" || typeof b !== "number") {
    throw new Error("Both arguments must be numbers.");
  }
  return a + b;
}
