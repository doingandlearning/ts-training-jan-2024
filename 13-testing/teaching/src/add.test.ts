import { add } from "./add";
import { describe, test, expect } from "vitest";

describe("Happy path for add.ts", () => {
  test("will add numbers together", () => {
    // Arrange   : Given
    // Act       : When
    const result = add(1, 2);
    // Assert    : Then
    expect(result).toBe(3);
    expect(result); // chai
  });
});

describe("Unhappy path", () => {
  test("add throws for non numbers", () => {
    expect(() => add(1, "1" as any)).toThrow();
    expect(() => add(1, "1" as any)).toThrowErrorMatchingSnapshot(
      `[Error: Both arguments must be numbers.]`
    );
  });
});
