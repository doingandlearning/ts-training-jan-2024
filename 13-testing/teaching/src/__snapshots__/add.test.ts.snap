// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Unhappy path > add throws for non numbers > [Error: Both arguments must be numbers.] 1`] = `[Error: Both arguments must be numbers.]`;
